addon/components/basic-dialog.js

```diff
@@ -70,7 +70,8 @@ export default Component.extend({
       }
 
       // if the click is within the dialog, do nothing
-      if (document.querySelector(modalSelector).contains(target)) {
+      let modalEl = document.querySelector(modalSelector);
+      if (modalEl && modalEl.contains(target)) {
```

Comment:
This is _safer_, but I'm not sure it's quite _correct_. If there's no `modalEl`, it still runs `onClose`. My impression is that it should be
```js
let modalEl = document.querySelector(modalSelector);
if (!modelEl || modelEl.contains(target)) {
  return;
}

this.sendAction('onClose');
```